benchmark/scatter.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const fork = require('child_process').fork;
+const path = require('path');
+const CLI = require('./_cli.js');
+
+//
+// Parse arguments
+//
+const cli = CLI(`usage: ./node benchmark/scatter.js <filename>
+  Run the benchmark script <filename> many times and output the rate (ops/s)
+  together with the benchmark variables as a csv.
+
+  --runs 30              number of samples
+  --var  variable=value  set benchmark variable (can be repeated)
+`, {
+  arrayArgs: ['var']
+});
+
+if (cli.required.length !== 1) {
+  cli.abort(cli.usage);
+  return;
+}
+
+// Create queue from the benchmarks list such both node versions are tested
+// `runs` amount of times each.
+const filepath = path.resolve(cli.required[0]);
+const name = filepath.slice(__dirname.length + 1);
+const runs = cli.optional.runs ? parseInt(cli.optional.runs, 10) : 30;
+
+let printHeader = true;
+
+(function recursive(i) {
+  const child = fork(path.resolve(__dirname, filepath), cli.optional.var);
+
+  child.on('message', function(data) {
+    // print csv header
+    if (printHeader) {
+      const confHeader = Object.keys(data.conf).join('", "');
+      console.log(`"filename", "${confHeader}", "rate", "time"`);
```

Comment:
Here is an example from running `./node benchmark/scatter.js benchmark/arrays/var-int.js`, the confusing part is inserting `"n", "type"` into the header.

```
"filename", "n", "type", "rate", "time"
"arrays/var-int.js", 25, Array, 73.73121801182165, 0.339069402
"arrays/var-int.js", 25, Buffer, 93.95908515556611, 0.266073259
"arrays/var-int.js", 25, Int8Array, 86.30949711522933, 0.289655262
"arrays/var-int.js", 25, Uint8Array, 95.8215561613897, 0.260901628
"arrays/var-int.js", 25, Int16Array, 86.45278377064632, 0.289175188
"arrays/var-int.js", 25, Uint16Array, 89.49914833863428, 0.279332267
"arrays/var-int.js", 25, Int32Array, 90.51250166094967, 0.276204939
```